spec/enumerables_spec.rb

```diff
@@ -0,0 +1,157 @@
+# spec/enumerables_spec.rb
+require './enumerable' #=> add this
+
+describe Enumerable do
+  describe 'my_each' do
+    let(:arr) { [] }
+    it 'returns a new array by incrementing each element of the given array by 1' do
+      [1, 2, 3].my_each { |item| arr << item + 1 }
+      expect(arr). to eql([2, 3, 4])
+    end
+  end
+
+  describe 'my_each_with_index' do
+    it 'returns a Hash element from given list of string elements and block argument of hash item and index' do
+      hash = {}
+      expect(%w[cat dog wombat].my_each_with_index do |item, index|
+        hash[item] = index
+      end).to eql(%w[cat dog wombat])
+    end
+    it 'returns the new array of numeric elements by applying block definition to the item and index arguments' do
+      arr = []
+      expect([1, 2, 3].my_each_with_index do |item, index|
+        arr << item + index
+      end).to eql([1, 2, 3])
+    end
+  end
+
+  describe 'my_select' do
+    it 'returns a new array of elements whose modulus to 3 is zero from a range object' do
+      expect((1..10).my_select { |i| i % 3 == 0 }).to eql([3, 6, 9])
+    end
+    it 'retruns a new array of elements by selecting even numbers from an array of numerics' do
+      expect([1, 2, 3, 4, 5].my_select(&:even?)).to eql([2, 4])
+    end
+    it 'retruns a new array of elements by selecting even numbers from an array of numerics' do
+      expect([1, 2, 3, 4, 5].my_select(&:odd?)).to eql([1, 3, 5])
+    end
+    it 'retruns a new array of elements by selecting elements with the same class when argument is a class' do
+      expect([1, 'a', 'b', 4, 5].my_select { |num| num.is_a?(Numeric) }).to eql([1, 4, 5])
+    end
+    it 'returns a new array of symbols which satisfies a block argument from an array of symbol elements' do
+      expect(%i[foo bar].my_select { |x| x == :foo }).to eql([:foo])
+    end
+  end
```

Comment:
### Add negative scenarios 
- Add negative scenarios, example  if you added "creates object if all required attributes are passed" test method, also is necessary to add "does not create the object if required attributes are missing", because we need to test all possible scenarios